Example notebook for qutip tensorflow.

AGaliciaMartinez opened the pull request:
Example notebook for qutip-tensorflow showcasing how to optimize a unitary gate. 

The notebook has been written to work with colab. A badge has been included but I have not been able to test it properly yet. I need to merge the notebook first for that. 

I am not sure how well git diff will work with notebooks. Nevertheless, you can take a look at the notebook in my fork at the branch `example_notebook`.

AGaliciaMartinez committed:
Example notebook for qutip tensorflow.

AGaliciaMartinez committed:
Added colab link.

quantshah committed:
tidyup of the notebook

quantshah commented:
@AGaliciaMartinez I tidied up the notebook by getting all imports at the top and dividing the example in sections. Also removed redefining things and adding some more explanations. Please have a look.

This is extremely cool btw to be able to use qutip and tensor flow together in this seamless manner. The data layer + qutip-TensorFlow makes me very excited about various possibilities @hodgestar @jakelishman 

I look forward to adding some more examples and documentation.

hodgestar commented:
Reviewing here (since I can't open the notebook in the review page):

* The QuTiP auto_tidyup removal has landed now (qutip/qutip#1615) so we can remove that note now.
* It would be good to re-run with Matplotlib 3.3 to avoid all the matplotlib 3.4 errors in the output (although we should probably fix those in QuTiP itself too).

AGaliciaMartinez committed:
Suggested changes in PR

- One or two typos
- Removed line of code that sets autotidyup to False since relevant PR
is now merged
- Removed autoreload statements
- Executed notebook with matplotlib 3.3 to avoid warning messages.

AGaliciaMartinez commented:
Thank you for reviewing @quantshah and @hodgestar! I implemented the changes suggested and will merge once the checks finish.

AGaliciaMartinez committed:
Merge branch 'qutip:main' into example_notebook
